src/main/java/com/example/alaa/university/repository/UniversityRepository.java

```diff
@@ -1,4 +1,134 @@
 package com.example.alaa.university.repository;
 
-public class UniversityRepository {
+import com.example.alaa.university.domain.Address;
+import com.example.alaa.university.domain.Student;
+import com.example.alaa.university.domain.University;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementCreator;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.stereotype.Component;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Timestamp;
+import java.util.List;
+
+
+//import java.util.List;
+@Component
+public class UniversityRepository implements IUniversityRepository {
+    private final JdbcTemplate jdbcTemplateU;
+    private final RowMapper<University> rowMapperU;
+    private final IAddressRepository iAddressRepository;
+    private final IStudentRepository iStudentRepository;
```

Comment:
it should not be there. Alll the logic about complex remove, add, update should be done in service layer